src/stieltjes.jl

```diff
@@ -75,4 +75,165 @@ end
     D = T \ apply(*, log.(abs.(x .- x')), P)
     c = inv(2*kr.A)
     T[kr,:] * Diagonal(Vcat(2*convert(V,π)*c*log(c), 2c*D.diag.args[2]))
+end
+
+#################################################
+# ∫f(x)g(x)(t-x)^a dx evaluation where f and g in Legendre
+#################################################
+const PowKernelPoint{T, V, D, F} = BroadcastQuasiVector{T, typeof(^), Tuple{ContinuumArrays.AffineQuasiVector{T, V, Inclusion{V, D}, T}, F}}
+
+############
+# METHODS
+############
+
+# This will need re-adjustments now that the cached version exists, so I'm commenting it out for now
+
+# function dot(f::AbstractVector{T}, K::PowKernelPoint, g::AbstractVector{T}) where T
+#     (lf, lg) = (length(f),length(g))
+#     a = K.args[2]
+#     t = (K.args[1])[0.] # there must be something better than this? 
+#                         # maybe something in the spirit of (K.args[1]).args[1]?
+#     t<1 && error("t must be greater than 1.")
+#     (lf<∞) && (lg<∞) && return pointwisedot(f,g,a,t)
+#     ((lf<∞) || (lg<∞)) && error("TODO: Currently only both finite or both infinite.")
+#     # for now, seek naive convergence for infinite input.
+#     (i,conv1,conv2) = (0,1,2)
+#     while abs(conv1-conv2)>1e-15
+#         i = i+1
+#         conv1 = pointwisedot(f[1:i*20],g[1:i*20],a,t)
+#         conv2 = pointwisedot(f[1:2*i*20],g[1:2*i*20],a,t)
+#     end
+#     return conv2
+# end
+# function *(g::Adjoint, K::PowKernelPoint{<:Any,<:Any,<:ChebyshevInterval}, f::AbstractVector)
+#     return dot(g',K,f)
+# end
+
+
+############
+# IMPLEMENT CACHED VERSION
+############
+# Constructors support BigFloat and it's recommended to use them for high orders.
+mutable struct PowerLawIntegral{T, PP<:AbstractQuasiMatrix} <: AbstractCachedMatrix{T}
+    P::PP # OPs - only Legendre supported for now
+    a::T  # naming scheme follows (t-x)^a
+    t::T
+    data::Matrix{T}
+    datasize::Tuple{Int,Int}
+    array
+
+    function PowerLawIntegral{T, PP}(P::PP, a::T, t::T) where {T, PP<:AbstractQuasiMatrix}
+        new{T, PP}(P,a,t, pointwisecoeffmatrixdense(a,t,10),(10,10))
+    end
+end
+PowerLawIntegral(P::AbstractQuasiMatrix, a::T, t::T) where T = PowerLawIntegral{T,typeof(P)}(P,a,t)
+size(K::PowerLawIntegral) = (∞,∞) # potential to add maximum size of operator
+
+# data filling
+function _legendrepowerlaw_fill_data!(K::PowerLawIntegral, inds)
```

Comment:
Why this extra function?

Reply:
Just a leftover from the original structure I started with and can safely be removed.